apps/web/components/examples/blockchain/advanced-learning-platform.tsx

```
@@ -0,0 +1,756 @@
+          context: 'loadUserData'
+        }])
+      }
+    }
+
+    if (userId) {
+      loadUserData()
+    }
+  }, [userId, courseData, isAuthenticated, getUserProgress, getUserBadges, getOverallProgress, hasGraduationNFT, checkEligibility])
+
+  // Set up real-time event subscriptions
+  useEffect(() => {
```

Biome (2.1.2)

[error] This hook does not specify its dependency on checkForAutomaticBadges.

This dependency is being used here, but is not specified in the hook dependency list.

Unsafe fix: Add the missing dependency to the list.


(lint/correctness/useExhaustiveDependencies)